tests/serverpod_test_server/test_integration/websockets/method_websockets/close_method_stream_command_test.dart

```diff
@@ -96,5 +96,77 @@ void main() {
         );
       });
     });
+
+    group('with a connected method stream that has a stream response', () {
+      late Completer<void> webSocketCompleter;
+      late Completer<void> delayedResponseClosed;
+
+      var endpoint = 'methodStreaming';
+      var method = 'delayedStreamResponse';
+      var connectionId = const Uuid().v4obj();
+
+      setUp(() async {
+        var delayedResponseOpen = Completer<void>();
+        delayedResponseClosed = Completer<void>();
+        webSocketCompleter = Completer<void>();
+
+        webSocket.stream.listen((event) {
+          var message = WebSocketMessage.fromJsonString(event);
+          if (message is OpenMethodStreamResponse) {
+            if (message.connectionId == connectionId)
+              delayedResponseOpen.complete();
+          } else if (message is CloseMethodStreamCommand) {
+            if (message.connectionId == connectionId)
+              delayedResponseClosed.complete();
+          }
+        }, onDone: () {
+          webSocketCompleter.complete();
+        });
+
+        webSocket.sink.add(OpenMethodStreamCommand.buildMessage(
+          endpoint: endpoint,
+          method: method,
+          args: {'delay': 10},
+          connectionId: connectionId,
+        ));
+
+        await expectLater(
+          delayedResponseOpen.future.timeout(Duration(seconds: 5)),
+          completes,
+          reason: 'Failed to open all method streams with server.',
+        );
```

Comment:
Question: This seem like the wrong place to put an expect. The setup section should not validate / test things. Just create dependencies for the test.

Reply:
I wasn't sure if an assert would operate correctly with async methods. I basically want to assert that the future completes.

Reply:
Fixed with throwing asserions instead. This should give us a clearer distinction between a failing test and a failed setup.